_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+```
+
+You should see the error 
+```
+oc get po -n mcourcy-pacman 
+NAME                              READY   STATUS                 RESTARTS        AGE
+failure-f8698c76d-9drv5           0/1     CreateContainerError   0               43s
+pacman-79cbb686d7-fntgg           1/1     Running                1 (28d ago)     28d
+pacman-mongodb-67fffc6787-7fzgx   1/1     Running                2 (2d20h ago)   28d
+```
+
+Now take a backup, Kasten will wait for the deployment to be ready and after a while it will fail the backup.
+
+![Failed backup](images/posts/2024-01-12-kasten-openshift-monitoring/failed-backup.png)
+
+
+# Make sure the failed actions are now populated in openshift-monitoring 
+
+I will query the failed action from openshift monitoring as we did before but I will use a criteria `policy='mcourcy-pacman-backup'` to narrow my search.
+
+```
+query=catalog_actions_count{status='failed',policy='mcourcy-pacman-backup'}
+```
+
+Let's repeat the complete action to query from Thanos:
+
+```
+SECRET=`oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'`
+TOKEN=`echo $(oc get secret $SECRET -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d`
+THANOS_QUERIER_HOST=`oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'`
+curl -X GET -kG "https://$THANOS_QUERIER_HOST/api/v1/query?" --data-urlencode "query=catalog_actions_count{status='failed',policy='mcourcy-pacman-backup'}" -H "Authorization: Bearer $TOKEN" |jq
+```
+
+you should get a similar output : 
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "__name__": "catalog_actions_count",
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "mcourcy-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "mcourcy-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "failed",
+          "type": "backup"
+        },
+        "value": [
+          1705076841.476,
+          "2"
+        ]
+      }
+    ]
+  }
+}
+```
+
+check the timestamp `1705076841.476,` and convert it in a human readable format.
+
+```
+date -d @1705076841.476
+Fri Jan 12 17:27:21 CET 2024
+```
+
+If you are on mac 
+```
+brew install coreutils
+gdate -d @1705076841.476
+Fri Jan 12 17:27:21 CET 2024
+```
+
+The date and time should be the date and time of the failed backup. The number `2` is because we have a failed backup action and also a failed runaction because we trigger the backup action from a policy.
```

Comment:
```suggestion
The date and time should be the date and time of the failed backup. The number `2` in the earlier response is because we have a failed backup action and also a failed runaction because we trigger the backup action from a policy.
```